charon-ml/src/TypesUtils.ml

```diff
@@ -169,6 +169,40 @@ let trait_instance_id_as_trait_impl (id : trait_instance_id) :
   | TraitImpl impl_ref -> (impl_ref.id, impl_ref.generics)
   | _ -> raise (Failure "Unreachable")
 
+let is_signed (int_ty : integer_type) =
+  match int_ty with
+  | Isize | I8 | I16 | I32 | I64 | I128 -> true
+  | Usize | U8 | U16 | U32 | U64 | U128 -> false
+
+let is_in_bounds (int_ty : integer_type) (v : Z.t) =
+  match int_ty with
+  | Isize -> Z.fits_int v
+  | I8 -> (v >= Z.(~- (~$0x80))) && v <= Z.of_int 0x7F
+  | I16 -> (v >= Z.(~- (~$0x8000))) && v <= Z.of_int 0x7FFF
+  | I32 -> Z.fits_int32 v
+  | I64 -> Z.fits_int64 v
+  | Usize -> Z.fits_int64_unsigned v
+  | U8 -> v >= Z.zero && v <= Z.of_int 0xFF
+  | U16 -> v >= Z.zero && v <= Z.of_int 0xFFFF
+  | U32 -> Z.fits_int32_unsigned v
+  | U64 -> Z.fits_int64_unsigned v
+  | I128 | U128 ->
+      true (* Only true if derived from anything representable in rust *)
+
+let max_of (int_ty : integer_type) =
+  (* FIXME: Not sure whether all of these values are even representable before being made into Z.t. *)
+  match int_ty with
+  | I8 -> Z.of_int 0x7FF
+  | I16 -> Z.of_int 0x7FFF
+  | I32 -> Z.of_int32 Int32.max_int
+  | Isize | I64 -> Z.of_int64 Int64.max_int
```

Comment:
This implicitly assumes a 64 bits target. I'm not sure how do deal with this well, maybe we can just codument this assumption on the function.

Reply:
`i64` is big enough to contain both a `i32` or a `i64` so that's ok. It's just that on 32 bit platforms the max value stored in it will be `i32::MAX`.